azure-kit-starter/src/main/java/com/vaadin/azure/starter/ui/ClusterSupport.java

```diff
@@ -0,0 +1,245 @@
+package com.vaadin.azure.starter.ui;
+
+import javax.servlet.http.Cookie;
+import java.io.Serializable;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.vaadin.flow.component.Component;
+import com.vaadin.flow.component.ComponentEvent;
+import com.vaadin.flow.component.ComponentEventListener;
+import com.vaadin.flow.component.DomEvent;
+import com.vaadin.flow.component.Tag;
+import com.vaadin.flow.component.dependency.JsModule;
+import com.vaadin.flow.internal.CurrentInstance;
+import com.vaadin.flow.server.RequestHandler;
+import com.vaadin.flow.server.ServiceInitEvent;
+import com.vaadin.flow.server.VaadinRequest;
+import com.vaadin.flow.server.VaadinResponse;
+import com.vaadin.flow.server.VaadinServiceInitListener;
+import com.vaadin.flow.shared.Registration;
+
+/**
+ * Cluster support for Vaadin applications. This component allows receiving
+ * events from the cluster running the Vaadin application.
+ */
+public class ClusterSupport implements VaadinServiceInitListener {
+
+    private static final long serialVersionUID = 1L;
+
+    /** Configuration property name. */
+    public static final String PROPERTY_APP_VERSION = "app.version";
+    public static final String ENV_APP_VERSION = "APP_VERSION";
+
+    /* Control cookie name. */
+    public static final String CURRENT_VERSION_COOKIE = "app-version";
+    public static final String UPDATE_VERSION_COOKIE = "app-update";
+    public static final String STICKY_CLUSTER_COOKIE = "INGRESSCOOKIE";
+
+    /* Global listener for version changes. */
+    private SwitchVersionListener switchVersionListener;
+
+    private String appVersion;
+
+    @Tag("version-notificator")
+    @JsModule("./components/version-notificator.ts")
+    public static class VersionNotificator extends Component {
+        private static final long serialVersionUID = 1L;
+
+        VersionNotificator(String current, String update) {
+            getElement().setProperty("currentVersion", current);
+            getElement().setProperty("updateVersion", update);
+        }
+
+        public Registration addSwitchVersionEvent(
+                ComponentEventListener<SwitchVersionEvent> listener) {
+            getLogger().debug("Adding listener for SwitchVersionEvent.");
+            return addListener(SwitchVersionEvent.class, listener);
+        }
+
+        @DomEvent("load-version")
+        public static class SwitchVersionEvent
+                extends ComponentEvent<VersionNotificator> {
+            private static final long serialVersionUID = 1L;
+
+            public SwitchVersionEvent(VersionNotificator source,
+                    boolean fromClient) {
+                super(source, fromClient);
+            }
+        }
+    }
```

Comment:
This class and its public methods needs JavaDoc, and also the inner event class.

Reply:
Added the JavaDoc.